packages/next-plugin-treat/index.js

```diff
@@ -0,0 +1,21 @@
+const TreatPlugin = require('treat/webpack-plugin');
+
+module.exports = (pluginOptions = {}) => (nextConfig = {}) => {
+  return Object.assign({}, nextConfig, {
+    webpack(config, options) {
+      config.plugins.push(
+        new TreatPlugin(
+          Object.assign({}, pluginOptions, {
```

Comment:
How does mini-extract get applied?

Reply:
It turns out that Next.js applies it without intervention. Not sure about how, but both debug and production builds work properly. I’ve tried to add mini-extract explicitly, but it always caused trouble and resulted in bad renders.

Reply:
You need to set `outputLoaders` though. Otehrwise treat will default to using `style-loader`.

Reply:
You need outputLoaders @kripod - think it only worked because of style-loader.

What you had before, was perfect.

Reply:
Static CSS is now extracted, but only in production, [as Next.js does it internally](https://github.com/zeit/next.js/blob/a81d7620a40b2006cb6726236402fc3f82f65ab8/packages/next/build/webpack/config/blocks/css/index.ts#L298-L320).